### ISSUE TITLE
Problem with gt-10

### ISSUE THREAD

**Issue by sikejsudjek** [snhirsch/katana-midi-bridge#9]
Hi, thanks for this great utility!
I've got the volume control working, but not save/restore patches when using a boss gt-10. Looking at the midi data it outputs it sends out bank change cc messages along with the program change. Would this be why I can't get the save patch/recall patch function working ? 

Each time I change patch on the gt-10 it sends the following :
pc # number
cc#0 value  bank select msb
cc#32 bank select lsb 

Unfortunately the gt-10 is fairly limited in its midi implementation and there doesn't seem to be any way to switch this off. Could the code be adjusted to ignore bank change cc#0 and cc#32 ? 

I have checked I am correctly sending out cc 3 with a value of 127 within 3 seconds (by using three different ctr pedals because latching/unlatching one pedal sends cc 3 value 127 and cc 3 value 0 !)

**Comment by snhirsch** [comment]
I guess I"m not sure what you are trying to accomplish.  My controller code was written for a Katana, not a GT-10.  Can you elaborate a bit on your setup?

**Comment by sikejsudjek** [comment]
Hi Snhirsch,

Sorry about the confusion,

I'm just using a gt-10 with a katana 100 watt combo as a midi controller
and trying to use the bridge to load more presets into the katana. The
reason for this is that the gt10's preamps are poor in comparison, and I
can only have two preamps set up on the katana and switch between these two
via the gt10 cable select method. Hence having the bridge save and load
more options I can use more of the katana's superior preamps/patches - and
select them with the gt10 via program changes.

The gt10 is sending  cc70 via its pedal into the katana via the bridge and
controls the volume so I know the bridge is set up and running ok. Also
program change numbers are correctly coming out of the gt10 and switching
the katana's tone presets with pc# 1-4. However I can't get the save/load
feature of the bridge to work into the katana.

I think the problem's the way the gt-10 sends out midi commands ? I plugged
it into a sequencer and recorded the midi data.  It sends cc 0 and cc 32 at
the same time it sends a pc number (for bank select). There doesn't seem to
be any way of disabling these extra cc messages. Are they disabling the
save feature ? Not sure if other boss multifx pedals work this way but this
might be a problem if so.

Thanks for the reply
Best wishes
Andy

**Comment by snhirsch** [comment]
Yes, if the GT-10 insists on sprinkling other messages around the CCs you won't be able to trigger a save.  The state machine code expects (3) CC#3 = 127 in two seconds.  If it sees anything else within that time period it resets and goes back to regular mode.  It should be fairly simple to ignore the CC#0 and CC#32 messages.  I will try to code something up this week, but will have to rely on you to test.

**Comment by sikejsudjek** [comment]
Thanks snhirsch,

No problem with testing the change. I had a look at the boss gt-100 manual
and it appears to do the same as the gt-10 by sending out cc messages with
the pc. Maybe this is how most boss pedals deal with program banks of over
the 127 limit ? Annoyingly they don't seem to have provided any way of
turning the cc#'s off.

Many thanks
Andy

**Comment by gumtown** [comment]
FYI: most Roland/Boss midi products which manage more than 128 patches send Bank select cc# + control change cc# + program number. 
This includes all the GT range, all the GR range, VG range , synths/keyboards, and more. 
It is a good reason to not use cc#00 and cc#32 for controllers, the software will have either ignore them, or include them as one package if the pause between data byte packages are <40ms (approximate).
